### ISSUE TITLE
Code Integrity check failed because of EXTRA_FILE after Update from 15.0.8 to 16.0.3

### ISSUE THREAD

**Issue by Rayn0r** [nextcloud/server#16375]
This one may be related to #16229.

### Steps to reproduce
1. Open Admin page: https://server/settings/integrity/failed

### Expected behaviour
Integrity check should not fail.

### Actual behaviour
There are two files in the integrity check which do not seem to belong here. The `README.md` must have come with the app. I installed it via the UI. I have no problem deleting it...

The `.rnd` seems to be more persistent... After deleting it, it will re-appear after a certain period of time. I have no idea where it comes from. It has been there before the upgrade, but the integrity check had not complained about it. It however bothered the Updater, which refused to install 16.0.3 before I deleted the file. After the upgrade the `.rnd` file has re-appeared and is now reported.

### Server configuration

**Operating system**: Ubuntu 18.04

**Web server:** 2.4.29-1ubuntu4.6

**Database:** MariaDB 10.3.16+maria

**PHP version:** 7.2.19-0ubuntu0.18.04.1

**Nextcloud version:** 16.0.3

**Updated from an older Nextcloud/ownCloud or fresh install:** Upgrade from 15.0.8

**Where did you install Nextcloud from:** used the updater

**Signing status:**
<details>
<summary>Signing status</summary>

```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- EXTRA_FILE
		- .rnd
- files_rightclick
	- EXTRA_FILE
		- README.md

Raw output
==========
Array
(
    [core] => Array
        (
            [EXTRA_FILE] => Array
                (
                    [.rnd] => Array
                        (
                            [expected] => 
                            [current] => 8b217381e027ab5da256184c9977df618e50f1256a140e4b22dfc13ffbeac9b55c499c2a08e925294e8ebca85f7c2525781db8fef83309b9d2271840a15a48dc
                        )

                )

        )

    [files_rightclick] => Array
        (
            [EXTRA_FILE] => Array
                (
                    [README.md] => Array
                        (
                            [expected] => 
                            [current] => cf73849388838de5037624e53303618579b164ec69fd55834ce4c5332ae33f778839e36298cbcc8ede90620927e076dbb8883479754e4c09b0491b15bf7653f3
                        )

                )

        )

)
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
sudo -u www-data php occ app:list
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud.andre-weidemann.de",
            "home.andre-weidemann.de",
            "192.168.3.43",
            "server"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/nextcloud.andre-weidemann.de\/",
        "htaccess.RewriteBase": "\/",
        "dbtype": "mysql",
        "version": "16.0.3.0",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "forcessl": true,
        "forceSSLforSubdomains": true,
        "tempdirectory": "\/tmp",
        "theme": "",
        "maintenance": false,
        "login_form_autocomplete": false,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0,
            "dbindex": 0
        },
        "loglevel": 0,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "singleuser": false,
        "updater.release.channel": "stable",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "enable_previews": true,
        "preview_max_x": 2048,
        "preview_max_y": 2048,
        "preview_jpeg_quality": 75,
        "mysql.utf8mb4": true,
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** FF

**Operating system:** Win10

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Nothing to contribute to the problem
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Nothing to contribute to the problem
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Nothing to contribute to the problem
```
</details>

**Comment by enoch85** [comment]
Integrity check fails in the Nextcloud VM as well. It didn't happen in 16.0.1 but started to happen when upgrading from 16.0.1/16.0.2 --> 16.0.3.

More info under "Known issues" here: https://github.com/nextcloud/vm/releases/tag/16.0.3

**Comment by Rayn0r** [comment]
@enoch85 
I read the "Known issues" article, and must say that the files mentioned there are not the ones the integrity check complains about here.
Disabling the Integrity check is not an option. I'd rather live with the message than forget to enable it again after the next upgrade.
I'm most curious of where the `.rnd` file comes from... It re-appeared once more:
```
-rw-------   1 www-data www-data  1024 Jul 12 17:00 .rnd
```

**Comment by enoch85** [comment]
@Rayn0r Sorry, reading again I see now that this is a different issue.

**Comment by MorrisJobke** [comment]
rnd file is solved by #16492
